SpatialGDK/Source/SpatialGDK/Public/Utils/SpatialActorUtils.h

```diff
@@ -2,18 +2,43 @@
 
 #pragma once
 
-#include "CoreMinimal.h"
+#include "EngineClasses/SpatialNetConnection.h"
 
 #include "Components/SceneComponent.h"
+#include "Containers/Array.h"
+#include "Containers/UnrealString.h"
 #include "Engine/EngineTypes.h"
-#include "EngineClasses/SpatialNetConnection.h"
 #include "GameFramework/Actor.h"
 #include "GameFramework/Controller.h"
 #include "GameFramework/PlayerController.h"
+#include "Math/Vector.h"
 
 namespace SpatialGDK
 {
 
+// We need the optional ActorToStopIteratingAt argument because for ownership-based Actor sets we want to find a hierarchy
+// assuming the future deletion of a given Actor. This is because Actor deletions are broadcast before SetOwner is called
+// which then updates the ownership hierarchy.
+inline TArray<AActor*> GetOwnershipHierarchyPath(const AActor* Actor, const AActor* ActorToStopIteratingAt = nullptr)
```

Comment:
So if I were to use this function, I'd only specify `ActorToStopIteratingAt` if that Actor has just been deleted? Perhaps we can just check for pending kill?

Reply:
Unfortunately not, OnDestroyed gets broadcast before MarkPendingKill is called